strings/base_coroutine_foundation.h

```diff
@@ -801,4 +801,43 @@ WINRT_EXPORT namespace winrt
         impl::check_status_canceled(shared->status);
         co_return shared->result.GetResults();
     }
+
+    template <typename Delegate>
+    struct async_event : impl::event_base<Delegate>
+    {
+        async_event() = default;
+
+        template<typename...Arg>
+        Windows::Foundation::IAsyncAction operator()(Arg const... args)
+        {
+            auto targets = this->make_temp_targets();
+
+            if (!targets)
+            {
+                co_return;
+            }
+
+            std::vector<Windows::Foundation::IAsyncAction> workers;
+            workers.reserve(targets->size());
+
+            for (auto&& element : *targets)
+            {
+                workers.push_back(invoke(element, args...));
+            }
+
+            for (auto&& element : workers)
+            {
+                co_await element;
+            }
+        }
+
+    private:
+
+        template <typename... Args>
+        Windows::Foundation::IAsyncAction invoke(Delegate const& element, Args const&... args)
+        {
+            co_await resume_background();
+            impl::invoke(element, args...);
```

Comment:
But this doesn't have the "auto-revoke on disconnected" behavior that the sync event supports. Shouldn't it return `IAsyncOperation<bool>` and `co_return impl::invoke(element, args...)` and then the `co_await` loop up above would check if the `invoke` failed and if so, revoke the token for the element?

Reply:
The trouble is I can't easily guarantee the event source still lives.

Reply:
It's probably safe because the "sender" is likely the strong reference we need to make it so, but there's no way to guarantee this.